lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -945,6 +983,21 @@ class AndroidTextEditingStrategy extends GloballyPositionedTextEditingStrategy {
 class FirefoxTextEditingStrategy extends GloballyPositionedTextEditingStrategy {
   FirefoxTextEditingStrategy(HybridTextEditing owner) : super(owner);
 
+  @override
+  void initializeTextEditing(
+    InputConfiguration inputConfig, {
+    @required _OnChangeCallback onChange,
+    @required _OnActionCallback onAction,
+  }) {
+    super.initializeTextEditing(inputConfig,
+        onChange: onChange, onAction: onAction);
+    if (hasAutofillGroup) {
+      placeForm();
+    } else {
+      domRenderer.glassPaneElement.append(domElement);
+    }
```

Comment:
It looks like we are repeating this logic in multiple places. Should this `if` be inside `placeForm` instead?

Reply:
This class is calling `placeForm` from the parent `DefaultTextEditingStrategy`. Yet not all children of DefaultTextEditingStrategy have the same implementation. Ex: `SemanticsTextEditingStrategy`